packages/bus-rdf-resolve-quad-pattern/lib/ActorRdfResolveQuadPattern.ts

```diff
@@ -108,6 +117,8 @@ IActorRdfResolveQuadPatternOutput> {
 export type IDataSource = string | RDF.Source | {
   type?: string;
   value: string | RDF.Source;
+  username?: string;
```

Comment:
I would not mention `username` and `password` here explicitly. A more extensible way of working would be to use a `sourceContext?: ActionContext` field, which contains additional context entries specifically for this source (explain this as a jsdoc comment as well).

Reply:
Calling it just `context` is probably fine as well.